crates/fuel-core/src/database/storage.rs

```diff
@@ -51,7 +55,7 @@ where
     }
 }
 
-// TODO: After https://github.com/FuelLabs/fuel-vm/pull/679 implement it only for `feature = "test-helpers"`
+#[cfg(feature = "test-helpers")]
```

Comment:
Why is this just a test impl? `StorageMutate` and `Database` seem like implementation concepts, not just test concepts. Is it more to do with the constraints on the blanket impl?

Reply:
> The StorageMutate is only implemented for the StorageTransaction, and the Storage should only implement read methods(and only one mutate commit_changes method).
I just read this in the issue. Probably the answer to my question.